ocean_lib/models/v4/test/test_swap_fees.py

```diff
@@ -0,0 +1,3906 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+import pytest
+from web3 import exceptions
+import math
+from ocean_lib.models.v4.dispenser import DispenserV4
+from ocean_lib.models.v4.erc20_token import ERC20Token
+from ocean_lib.models.v4.bpool import BPool
+from ocean_lib.models.v4.side_staking import SideStaking
+from ocean_lib.models.v4.erc721_factory import ERC721FactoryContract
+from ocean_lib.models.v4.erc721_token import ERC721Token
+from ocean_lib.models.v4.models_structures import ErcCreateData, PoolData
+from ocean_lib.web3_internal.constants import ZERO_ADDRESS
+from tests.resources.helper_functions import get_address_of_type
+
+
+def deploy_erc721_token(config, web3, factory_deployer_wallet, manager_wallet):
```

Comment:
I have modified `deploy-erc721_erc20` helper function in order to return an erc721 if the erc20 wallet is missing and when that PR is merged we can remove this function and use that one for avoiding code duplication.